Documentation/WritingDocForExtension/Migrate.rst

```diff
@@ -32,32 +32,222 @@ a migration tool. This tool can be used to automatically create a
+documentation.
+
+Previously it was also used to define a list of utilized directives/roles.
+
+You can either safely remove that file, or add your fixed text to it.
+
+Most official documentation uses this as the stub of the file:
+
+..  code-block:: text
+    :caption: /Documentation/Includes.rst.txt
+
+    ..  You can put central messages to display on all pages here
+
+4. Remove the entry `genindex` from :file:`Index.rst` (Index/Glossary)
+----------------------------------------------------------------------
+
+If you previously had a :file:`genindex.rst` file, this optional index
+(or glossary) was rendered as a page through an entry in the file :file:`Index.rst`
+like this:
+
+..  code-block:: text
+    :caption: /Documentation/Index.rst
+    :emphasize-lines: 17
+
+    **Table of Contents:**
+
+    .. toctree::
+       :maxdepth: 2
+       :titlesonly:
+
+       Introduction/Index
+       Installation/Index
+       Details/Index
+
+    .. Meta Menu
+
+    .. toctree::
+       :hidden:
+
+       Sitemap
+       genindex
+
+Remove the entry `genindex` from the list.
+
+..  hint::
+
+    See :ref:`migrate-glossary` for details about the future of the
+    index (glossary) generation.
+
+5. Prevent code-snippets with :file:`.rst` extension
+----------------------------------------------------
+
+All files ending in :file:`.rst` will be interpreted by the new rendering,
+and every file that is just a code snippet placed in an external file
+should be renamed to use a :file:`.rst.txt` extension instead.
+
+Recommendations
+===============
+
+The following list is not a requirement to utilize the new rendering, but
+follows "best practice" to make the most of your documentation project.
+
+..  _migrate-to-makefile:
+
+Add a :file:`Makefile` to your project
+--------------------------------------
+
+A :file:`Makefile` is a simple command line runner configuration file, which requires
+the utility `GNU make <https://www.gnu.org/software/make/manual/make.html>`_
+to be available on your Operating System.
+
+This allows you to perform shortcuts to render your documentation instead
+of typing a long :bash:`docker run...` or :bash:`podman run...` command:
+
+..  code-block:: shell
+
+    make docs
+
+For inspiration, check out the :file:`Makefile` of the main PHP-based rendering
+repository:
+
+https://github.com/TYPO3-Documentation/render-guides/blob/main/Makefile
+
+A small example :file:`Makefile`:
+
+..  literalinclude:: _Makefile
+    :caption: /Makefile
+
+..  _migrate-to-testing-workflow:
+
+Activate automatic testing in your project
+------------------------------------------
 
 It is recommended to use an automatic workflow on GitHub Or Gitlab to
 ensure the extension's documentation renders without warnings.
 
-..  todo: Add a link to the according chapter
+An example workflow on GitHub would be established via this file in
+:file:`.github/actions/documentation.yml`:
+
+..  code-block:: yaml
```

Comment:
Please move it to a separate file and use literalinclude. This way, syntax highlighting is available in the IDE and errors might be spotted better.

Reply:
Done